tests/rados/test_scrub_chunk_max.py

```diff
@@ -0,0 +1,173 @@
+    # CEPH-10792
+    Test to verify PG scrub is within the 'osd_scrub_chunk_max' limit
+    1. Create a replicated pool with single pg
+    2. Grab the default value of 'osd_scrub_chunk_max'
+    3. Set 'osd_scrub_chunk_max' to a low value, preferably lower than
+    the default. e.g. 5
+    4. Write few objects(>osd_scrub_chunk_max) to the created pool
+    5. Start deep-scrub of the pool with 1 pg having few objects
+    6. Wait for deep-scrub to complete and verify primary OSD logs
+        expected to find logs having entries like below:
+            be_scan_list (0/5 5:18dee445:::obj-12:head deep)
+            be_scan_list (0/5 5:18dee445:::obj-12:head deep)
+    7. Delete the created pools
+    """
+    log.info(run.__doc__)
+    config = kw["config"]
+    rhbuild = config["rhbuild"]
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    rados_obj = RadosOrchestrator(node=cephadm)
+    mon_obj = MonConfigMethods(rados_obj=rados_obj)
+    pool_obj = PoolFunctions(node=cephadm)
+    client_node = ceph_cluster.get_nodes(role="client")[0]
+    installer_node = ceph_cluster.get_nodes(role="installer")[0]
+    pg_scrub_log_list = []
+    pool_name = "test_scrub_max"
+    log.info(
+        "Running test case to verify PG scrub is within the 'osd_scrub_chunk_max' limit"
+    )
+
+    try:
+        # create a replicated pool with single pg
+        rados_obj.create_pool(pool_name=pool_name, **{"pg_num_max": 1})
+
+        # retrieve default value of osd_scrub_chunk_max
+        org_scrub_chunk_max = mon_obj.get_config(
+            section="osd", param="osd_scrub_chunk_max"
+        )
+        log.info(f"Original value of osd_scrub_chunk_max is {org_scrub_chunk_max}")
+        assert int(org_scrub_chunk_max) > 5
+
+        # create 23 objects in the pool
+        pool_obj.do_rados_put(client=client_node, pool=pool_name, nobj=23)
+        # set value of osd_scrub_chunk_max as 5
+        mon_obj.set_config(
+            **{"section": "osd", "name": "osd_scrub_chunk_max", "value": 5}
+        )
+
+        time.sleep(10)  # blind wait to let ceph df stats account for all the objects
+        objs = rados_obj.get_cephdf_stats(pool_name=pool_name)["stats"]["objects"]
+        assert objs == 23
+
+        primary_osd = rados_obj.get_pg_acting_set(pool_name=pool_name)[0]
+        pool_pgid = rados_obj.get_pgid(pool_name=pool_name)[0]
+        pool_id = pool_obj.get_pool_id(pool_name=pool_name)
+        pool_pg_dump = rados_obj.get_ceph_pg_dump(pg_id=pool_pgid)
+
+        org_last_deep_scrub = pool_pg_dump["last_deep_scrub"]
+        org_last_deep_scrub_stamp = pool_pg_dump["last_deep_scrub_stamp"]
+        org_objects_scrubbed = int(pool_pg_dump["objects_scrubbed"])
+
+        log.info(f"last_deep_scrub: {org_last_deep_scrub}")
+        log.info(f"last_deep_scrub_stamp: {org_last_deep_scrub_stamp}")
+        log.info(f"objects_scrubbed: {org_objects_scrubbed}")
+
+        assert mon_obj.set_config(section="osd", name="debug_osd", value="20/20")
```

Comment:
Add a comment stating reason for the enhanced logging level and the log levels required.

Reply:
done